public/app/features/explore/QueryLibrary/QueryTemplatesTable/index.tsx

```diff
@@ -34,12 +36,33 @@ type Props = {
 };
 
 export default function QueryTemplatesTable({ queryTemplateRows }: Props) {
+  const styles = useStyles2(getStyles);
   return (
     <InteractiveTable
       columns={columns}
       data={queryTemplateRows}
       getRowId={(row: { index: string }) => row.index}
       pageSize={20}
+      className={styles.table}
     />
   );
 }
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  table: css({
+    'tbody tr': {
+      position: 'relative',
+      backgroundColor: theme.colors.background.secondary,
+      borderBottom: 'unset',
+      overflow: 'hidden', // Ensure the row doesn't overflow and cause scrollbars
+    },
+    /* Adds the pseudo-element for the lines between table rows */
```

Comment:
Tried an approach with `borderCollapse: 'collapse'` and setting border. This didn't work because the changes apply to headings too and there's a weird border on the top of the table. 
Also tried individually setting borders and margins on table rows but that didn't work either.